environment/src/test/java/com/sequenceiq/environment/environment/validation/EnvironmentValidatorServiceTest.java

```diff
@@ -324,4 +331,30 @@ private Environment aValidEnvirontmentWithParent() {
 
         return environment;
     }
+
+    @ParameterizedTest
+    @MethodSource("freeIpaCreationArguments")
+    void shouldValidateFreeIpaCreation(int instanceCountByGroup, int spotPercentage, boolean valid) {
+        FreeIpaCreationDto freeIpaCreationDto = FreeIpaCreationDto.builder()
+                .withInstanceCountByGroup(instanceCountByGroup)
+                .withAws(FreeIpaCreationAwsParametersDto.builder()
+                        .withSpot(FreeIpaCreationAwsSpotParametersDto.builder()
+                                .withPercentage(spotPercentage)
+                                .build())
+                        .build())
+                .build();
+
+        ValidationResult validationResult = underTest.validateFreeIpaCreation(freeIpaCreationDto);
+        assertEquals(!valid, validationResult.hasError());
+    }
+
+    private static Stream<Arguments> freeIpaCreationArguments() {
+        return Stream.of(
+                Arguments.of(1, 0, true),
+                Arguments.of(1, 100, true),
+                Arguments.of(1, 50, false),
+                Arguments.of(2, 50, true)
```

Comment:
can you please add 2 more test ? 
Arguments.of(2, 100, true)
Arguments.of(2, 0, true)